ouroboros-consensus-test/test-consensus/Test/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -283,9 +299,12 @@ runChainSync securityParam (ClientUpdates clientUpdates)
                 readTVar varClientState
           , getIsInvalidBlock = return $
               WithFingerprint (const Nothing) (Fingerprint 0)
+
+          , getIsFetched = error "this test does not involve low-density chains"
+          , getBlock     = error "this test does not involve low-density chains"
           }
 
-        client :: StrictTVar m (AnchoredFragment (Header TestBlock))
+        client :: (CandidateFragment (Header TestBlock) -> m ())
```

Comment:
Comment explaining why this is in `m` and not in `STM m`. Crucially: it prevents us accidentally setting the variable without committing (in which case `BlockFetch` would actually see it).